src/analyzer/analyzergain.cpp

```diff
@@ -25,29 +26,41 @@ bool AnalyzerGain::initialize(TrackPointer tio, int sampleRate, int totalSamples
     if (loadStored(tio) || totalSamples == 0) {
         return false;
     }
-    m_bStepControl = m_pReplayGain->initialise((long)sampleRate, 2);
+
+    m_initalized = m_pReplayGain->initialise((long)sampleRate, 2);
     return true;
 }
 
 bool AnalyzerGain::loadStored(TrackPointer tio) const {
+    int version = m_pConfig->getValueString(
+            ConfigKey("[ReplayGain]", "ReplayGainVersion")).toInt();
     // WARNING: Do not fix the "analyser" spelling here since user config files
     // contain these strings.
-    bool bAnalyzerEnabled = (bool)m_pConfigReplayGain->getValueString(
-        ConfigKey("[ReplayGain]","ReplayGainAnalyserEnabled")).toInt();
-    if (tio->getReplayGain().hasRatio() || !bAnalyzerEnabled) {
-        return true;
+    bool analyzerEnabled = ((bool)m_pConfig->getValueString(
+            ConfigKey("[ReplayGain]", "ReplayGainAnalyserEnabled")).toInt()) &&
+            (version == 1);
+    bool reanalyse = m_pConfig->getValueString(
+            ConfigKey("[ReplayGain]", "ReplayGainReanalyze")).toInt();
+
+    if (analyzerEnabled) {
```

Comment:
The original condition is correct and behaves like the original version. 
I have added some comments. 
The issue is that the function name is somehow wrong. loadStored() makes sense for a beat grid analysis but not here. 

Here it should be named "ShouldRunAnalysis" or something. But since it is a virtual base class function, and I have not yet a good idea how it should work and be named, I would like to stick with the original name for now.